src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -223,15 +224,34 @@ public static ReadOnlySpan<char> GetFileNameWithoutExtension(ReadOnlySpan<char>
         /// Returns a cryptographically strong random 8.3 string that can be
         /// used as either a folder name or a file name.
         /// </summary>
-        public static unsafe string GetRandomFileName()
+        public static string GetRandomFileName()
         {
-            byte* pKey = stackalloc byte[KeyLength];
-            Interop.GetRandomBytes(pKey, KeyLength);
+            Span<char> buffer = stackalloc char[RandomFileNameLength];
+            bool result = TryGetRandomFileName(buffer, out int charsWritten);
 
-            const int RandomFileNameLength = 12;
-            char* pRandomFileName = stackalloc char[RandomFileNameLength];
-            Populate83FileNameFromRandomBytes(pKey, KeyLength, pRandomFileName, RandomFileNameLength);
-            return new string(pRandomFileName, 0, RandomFileNameLength);
+            Debug.Assert(charsWritten == RandomFileNameLength);
+            Debug.Assert(result);
+
+            return new string(buffer);
+        }
+
+        /// <summary>
+        /// Returns a cryptographically strong random 8.3 string that can be
+        /// used as either a folder name or a file name.
+        /// </summary>
+        public static unsafe bool TryGetRandomFileName(Span<char> destination, out int charsWritten)
+        {
+            charsWritten = 0;
+
+            if (destination.Length < RandomFileNameLength)
+                return false;
+
+            Span<byte> pKey = stackalloc byte[KeyLength];
+            Interop.GetRandomBytes(pKey);
+
+            Populate83FileNameFromRandomBytes(pKey, KeyLength, destination, destination.Length);
```

Comment:
>I'd change Populate83FileName to take Spans. It isn't perf critical, just don't want to add heap allocations.

To address this i changed also pinvoke to avoid trick to convert Span to pointer. This changes maybe are out of subject of this PR, if so can you tell me the correct way to convert Span to pointer?

Reply:
`fixed (byte* p = &span.GetPinnableReference())`

Once we get a C# compiler update, it can be just:

`fixed (byte* p = &span)`

Reply:
You need https://github.com/dotnet/coreclr/pull/17504

Reply:
As I suspected